ChildMindInstitute/mindlogger-admin
# feat: Revised PDP - About Participant (M2-7767)

- [ ] ~~Tests for the changes have been added~~ **(moved to separate task)**

### 📝 Description




🔗 [Jira Ticket M2-7767](https://mindlogger.atlassian.net/browse/M2-7767)

This PR introduces the new tabbed view of the Participant Details Page (PDP), and specifically the first tab of that screen, "About Participant". Changes include:

- Create navigation infrastructure to display this new tabbed view when the `enableActivityAssign` feature flag is enabled, and the original PDP when disabled
- Add support for new endpoint `/activities/applet/${appletId}/target/${subjectId}`
- Flesh out "About Participant" view, with a reusable hook, `useAssignmentsTab`, that can be reused by the "By Participant" view (assign, unassign, navigate to data, export data, unlock applet)
- Add empty state to About Participant view
- Slightly refactor `useAsync` to use a new, less awkward pattern for preserving the previous value when data is refetched (see [this commit message](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1936/commits/999927b33d400341b443a46f333302b90d960ab7) for details)

> [!TIP]
> Related changes have been grouped by commit, with commit messages containing more detailed descriptions as needed. Reviewing [commit-by-commit](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1936/commits) may be easier.

### 📸 Screenshots

https://github.com/user-attachments/assets/b37c0458-95f4-4761-9c30-0df6b8c1ca2f

### 🪤 Peer Testing

> [!NOTE] 
> This feature depends on the `enableActivityAssign` feature flag. This should already be enabled in dev/local environments. To control its value manually, add `features.enableActivityAssign = [true|false]` prior to the `return` statement [here in `useFeatureFlags`](https://github.com/ChildMindInstitute/mindlogger-admin/blob/e6ff173fc6aaf31da3e81d92ab4f59caa8eef6da/src/shared/hooks/useFeatureFlags.ts#L25-L26).

#### Test with feature flag enabled ✅

1. Add `features.enableActivityAssign = true` to [`useFeatureFlags`](https://github.com/ChildMindInstitute/mindlogger-admin/blob/e6ff173fc6aaf31da3e81d92ab4f59caa8eef6da/src/shared/hooks/useFeatureFlags.ts#L25-L26).
2. Please refer to the AC of [M2-7767](https://mindlogger.atlassian.net/browse/M2-7767) to ensure behaviour is in alignment.

#### Test with feature flag disabled 🚫

1. Add `features.enableActivityAssign = false` to [`useFeatureFlags`](https://github.com/ChildMindInstitute/mindlogger-admin/blob/e6ff173fc6aaf31da3e81d92ab4f59caa8eef6da/src/shared/hooks/useFeatureFlags.ts#L25-L26).
2. Load the PDP and ensure there are no regressions in behaviour.


[M2-7767]: https://mindlogger.atlassian.net/browse/M2-7767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
